### PR TITLE
Drop support for Swift 5.5

### PR DESCRIPTION
**Pull request by simonjbeaumont**
### Motivation:

Now Swift 5.8 is released, we are dropping support for 5.5, in line with the usual support window.

### Modifications:

- Remove `Package@swift-5.5.swift`.
- Remove `#if swift(>=5.6)` compiler guards.
- Update documentation and README.

### Result:

No functional change, but Swift 5.5 is no longer supported.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -48,20 +48,21 @@ declaring [SwiftNIO's Public API](https://github.com/apple/swift-nio/blob/main/d
...
-The most recent versions of SwiftNIO Transport Services support Swift 5.5.2 and newer. The minimum Swift version supported by SwiftNIO Transport Services releases are detailed below:
+The most recent versions of SwiftNIO Transport Services support Swift 5.6 and newer. The minimum Swift version supported by SwiftNIO Transport Services releases are detailed below:
 
 SwiftNIO Extras     | Minimum Swift Version
 --------------------|----------------------
 `1.0.0 ..< 1.11.0`  | 5.0
 `1.11.0 ..< 1.12.0` | 5.2
 `1.12.0 ..< 1.15.0` | 5.4
-`1.15.0 ...`        | 5.5.2
+`1.15.0 ..< 1.16.0` | 5.5.2
```
Comment (Lukasa) [inline README.md:64]: This is not _quite_ right: 1.16.0 does support 5.5.2. I struggled with this a bit myself in my other PRs and eventually put the equivalent of 1.16.1 in this spot.

File: Sources/NIOTransportServices/Docs.docc/index.md [hunk 2]
```diff
@@ -50,20 +50,21 @@ declaring [SwiftNIO's Public API](https://github.com/apple/swift-nio/blob/main/d
...
-The most recent versions of SwiftNIO Transport Services support Swift 5.5.2 and newer. The minimum Swift version supported by SwiftNIO Transport Services releases are detailed below:
+The most recent versions of SwiftNIO Transport Services support Swift 5.6 and newer. The minimum Swift version supported by SwiftNIO Transport Services releases are detailed below:
 
 SwiftNIO Extras     | Minimum Swift Version
 --------------------|----------------------
 `1.0.0 ..< 1.11.0`  | 5.0
 `1.11.0 ..< 1.12.0` | 5.2
 `1.12.0 ..< 1.14.0` | 5.4
-`1.14.0 ...`        | 5.5.2
+`1.15.0 ..< 1.16.0` | 5.5.2
```
Comment (Lukasa) [inline Sources/NIOTransportServices/Docs.docc/index.md:66]: Same note here.

File: README.md [hunk 3]
```diff
@@ -48,20 +48,21 @@ declaring [SwiftNIO's Public API](https://github.com/apple/swift-nio/blob/main/d
...
+`1.16.1 ...`        | 5.6
```
Comment (glbrntt) [inline README.md:65]: We treat dropping a Swift version as a minor change so this will be 1.17.0

**Review by glbrntt — COMMENTED** [review]
Looks good modulo one nit

File: README.md (hunk 3 above)
Comment (simonjbeaumont) [reply README.md]: Fixed this up in https://github.com/apple/swift-nio-transport-services/pull/172/commits/f1febb1934c0c945d2f9ed82dfff82d041e84347

File: README.md (hunk 1 above)
Comment (simonjbeaumont) [reply README.md]: https://github.com/apple/swift-nio-transport-services/pull/172/commits/f1febb1934c0c945d2f9ed82dfff82d041e84347

File: Sources/NIOTransportServices/Docs.docc/index.md (hunk 2 above)
Comment (simonjbeaumont) [reply Sources/NIOTransportServices/Docs.docc/index.md]: https://github.com/apple/swift-nio-transport-services/pull/172/commits/f1febb1934c0c945d2f9ed82dfff82d041e84347

**Review by Lukasa — APPROVED** [review]

**Comment by Lukasa** [comment]
@swift-server-bot add to allowlist

**Review by glbrntt — APPROVED** [review]

**Comment by simonjbeaumont** [comment]
@yim-lee Please can we remove the 5.5 CI for this project?

**Comment by yim-lee** [comment]
> Please can we remove the 5.5 CI for this project?

@simonjbeaumont Done

**Pull request merged**